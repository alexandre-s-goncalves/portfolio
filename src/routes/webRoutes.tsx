import React from 'react';
import {BrowserRouter, Routes, Route} from 'react-router-dom';
import {About} from 'pages/about/about';
import {Home} from 'pages/home';
import {Projects} from 'pages/projects/projects';
import {Skills} from 'pages/skills/skills';
import {NotFound} from 'pages/notFound';
import {WebRoutesContainer, Content} from './webRoutes.styles';
import {Header} from 'components/header';
import {useTheme} from 'context/ThemeContext';
import {Footer} from 'components/footer';

export const WebRoutes = () => {
  const {themeDark} = useTheme();
  return (
    <BrowserRouter>
      <WebRoutesContainer data-testid="web-routes-container" $backgroundblack={themeDark}>
        <Header />
        <Content data-testid="web-routes-content">
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/skills" element={<Skills />} />
            <Route path="/projects" element={<Projects />} />
            <Route path="/about" element={<About />} />
            <Route path="*" element={<NotFound />} />
          </Routes>
        </Content>
        <Footer />
      </WebRoutesContainer>
    </BrowserRouter>
  );
};
