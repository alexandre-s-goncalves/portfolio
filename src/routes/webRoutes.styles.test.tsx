import React from 'react';
import {render} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import * as S from './webRoutes.styles';

describe('webRoutes.styles', () => {
  describe('WebRoutesContainer', () => {
    test('should match the snapshot with default props', () => {
      const {container} = render(<S.WebRoutesContainer $backgroundblack={false} />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should apply light background when $backgroundblack is false', () => {
      const {getByTestId} = render(
        <S.WebRoutesContainer data-testid="container" $backgroundblack={false} />,
      );
      const container = getByTestId('container');
      expect(container).toHaveStyle('background-color: #ffffff');
    });

    test('should apply dark background when $backgroundblack is true', () => {
      const {getByTestId} = render(
        <S.WebRoutesContainer data-testid="container" $backgroundblack={true} />,
      );
      const container = getByTestId('container');
      expect(container).toHaveStyle('background-color: rgb(10, 10, 10)');
    });

    test('should render as main element', () => {
      const {container} = render(<S.WebRoutesContainer $backgroundblack={false} />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('MAIN');
    });

    test('should have flex layout with column direction', () => {
      const {getByTestId} = render(
        <S.WebRoutesContainer data-testid="container" $backgroundblack={false} />,
      );
      const container = getByTestId('container');
      expect(container).toHaveStyle('display: flex');
      expect(container).toHaveStyle('flex-direction: column');
    });

    test('should have full viewport height', () => {
      const {getByTestId} = render(
        <S.WebRoutesContainer data-testid="container" $backgroundblack={false} />,
      );
      const container = getByTestId('container');
      expect(container).toHaveStyle('height: 100vh');
    });
  });

  describe('Content', () => {
    test('should match the snapshot', () => {
      const {container} = render(<S.Content />);
      expect(container.firstChild).toMatchSnapshot();
    });

    test('should render as div element', () => {
      const {container} = render(<S.Content />);
      const element = container.firstChild as HTMLElement;
      expect(element.nodeName).toBe('DIV');
    });

    test('should have flex layout with full width and height', () => {
      const {getByTestId} = render(<S.Content data-testid="content" />);
      const content = getByTestId('content');
      expect(content).toHaveStyle('display: flex');
      expect(content).toHaveStyle('width: 100%');
      expect(content).toHaveStyle('height: 100%');
    });

    test('should have proper margins', () => {
      const {getByTestId} = render(<S.Content data-testid="content" />);
      const content = getByTestId('content');
      expect(content).toHaveStyle('margin-top: 116px');
      expect(content).toHaveStyle('margin-bottom: 70px');
    });

    test('should have vertical scroll', () => {
      const {getByTestId} = render(<S.Content data-testid="content" />);
      const content = getByTestId('content');
      expect(content).toHaveStyle('overflow-y: auto');
    });
  });
});
