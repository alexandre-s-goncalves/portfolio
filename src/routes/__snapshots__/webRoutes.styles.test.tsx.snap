// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webRoutes.styles Content should match the snapshot 1`] = `
.c0 {
  display: flex;
  width: 100%;
  height: 100%;
  overflow-y: auto;
}

<div
  class="c0"
/>
`;

exports[`webRoutes.styles WebRoutesContainer should match the snapshot with default props 1`] = `
.c0 {
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  min-height: 100vh;
  background-color: #ffffff;
}

<main
  class="c0"
/>
`;
